plugins/crd/crd_queue.go

```diff
@@ -49,39 +53,53 @@ func networkserviceWork(plugin *Plugin) {
 		// We define a function here to process a queue item, so that we can
 		// use 'defer' to make sure the message is marked as Done on the queue
 		func(key string) {
-			var obj interface{}
-			var err error
-
 			defer queueNS.Done(key)
 
 			// Attempt to split the 'key' into namespace and object name
 			namespace, name, err := cache.SplitMetaNamespaceKey(strKey)
 
 			if err != nil {
+				// This is a soft-error, retry up to QUEUE_RETRY_COUNT times
 				plugin.Log.Errorf("Error splitting meta namespace key into parts: %s", err.Error())
-				runtime.HandleError(fmt.Errorf("Error splitting meta namespace key into parts: %s", err.Error()))
-				// This is a soft-error, we merely want to forget it and mark it done on the queue
-				queueNS.Forget(key)
+				if err != nil {
```

Comment:
I cleaned this up while addressing one of @prateekgogia's comments, but did not push the new commit yet, so this will be cleaned up when I do that this morning.